### PR TITLE
Use updated `doc-validator` tool

### PR DESCRIPTION
**Pull request by jdbaldry**
#### What this PR does

Removes the old source code and uses an updated `doc-validator` tool that has been centralized so that it can be used by other source content repositories.

#### Checklist

- [na] Tests updated
- [ ] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by osg-grafana** [comment]
Can we open a separate issue for the documentation that describes what this does, or is this self explanatory enough based on code comments? (It very well might be.)

File: Makefile
Comment (pracucci) [inline Makefile:443]: Should we always run `docker pull grafana/doc-validator:latest` to ensure we always run on latest?

**Review by pracucci — APPROVED** [review]
Thanks! Exciting to see it used by other projects!

Could you update `docs/internal/contributing/how-to-write-documentation.md` too, please? Then once merged, we'll add `doc-validator` CI job to the list of required CI checks to merge a PR.

File: Makefile
Comment (jdbaldry) [reply Makefile:444]: Good idea, added in 034cec264

**Comment by jdbaldry** [comment]
Updated the internal docs in fc425954f

**Comment by jdbaldry** [comment]
@osg-grafana Addressed the nit in [f0cdfaf](https://github.com/grafana/mimir/pull/2307/commits/f0cdfafd60877c45ae9c06b381ad1d0a66da4931).

> Can we open a separate issue for the documentation that describes what this does, or is this self explanatory enough based on code comments? (It very well might be.)

The tool aims to be helpful in its output and I think should be self documenting in that way.
With that in mind, we should encourage people to raise issues if the error output is not easily understood and fixed.

**Comment by jhesketh** [comment]
> > Rather then vendoring this tool, would it make more sense to use github actions to run it as a container or the makefile to run it in docker?
> 
> I think the change to the checkout step names has been confusing here but this PR is doing exactly what you are suggesting.

Ah, my mistake! I was only skimming and read the PR intent completely wrong. I thought all those files were being vendored in, not being deleted. Will take a closer look+review tomorrow.

**Comment by pracucci** [comment]
_Merging so we can unblock the other PR from @jdbaldry._

**Pull request merged**
